### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by dammy001**

### CODE REVIEW

File: src/api/ApiIndex.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import { data as apiIndex, APIGroup } from './api.data'
 import { ref, computed } from 'vue'
 
-const query = ref('')
+const query = ref<string>('')
```
Comment (skirtles-code) [inline src/api/ApiIndex.vue:8]: I'm unclear what these changes to `ApiIndex.vue` and `api.data.ts` are. They seem to be unrelated to the typos.

File: src/guide/essentials/forms.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ The `v-model` directive helps us simplify the above to:
 In addition, `v-model` can be used on inputs of different types, `<textarea>`, and `<select>` elements. It automatically expands to different DOM property and event pairs based on the element it is used on:
 
 - `<input>` with text types and `<textarea>` elements use `value` property and `input` event;
-- `<input type="checkbox">` and `<input type="radio">` use `checked` property and `change` event;
+- `<input type="checkbox" />` and `<input type="radio" />` use `checked` property and `change` event;
```
Comment (skirtles-code) [inline src/guide/essentials/forms.md:35]: I don't think these were mistakes. Writing `<input>` without a slash is correct HTML 5 usage.

**Review by skirtles-code — CHANGES_REQUESTED** [review]
Thanks for the PR!

I've provided some feedback on your proposed changes below. I agree that the spelling mistake in `template-refs.md` needs fixing, as does the incorrect quote in `registration.md`. The other changes don't seem so clear cut to me

File: src/guide/essentials/forms.md (hunk 2 above)
Comment (dammy001) [reply src/guide/essentials/forms.md:35]: I know, it's just for consistency since we're self-closing input elements.

Comment (dammy001) [reply src/guide/essentials/forms.md:35]: @skirtles-code

Comment (skirtles-code) [reply src/guide/essentials/forms.md:35]: If we're going to make them consistent then I think we should consistently not include the slashes.

Comment (dammy001) [reply src/guide/essentials/forms.md:35]: @skirtles-code alright

File: src/api/ApiIndex.vue (hunk 1 above)
Comment (dammy001) [reply src/api/ApiIndex.vue:8]: @skirtles-code yeah, not related to the typos. It's just basically adding typing to query `ref` and destructuring the parent array on line 74.

File: src/guide/essentials/forms.md [hunk 3]
```diff
@@ -168,13 +168,13 @@ export default {
 <div class="demo">
   <div>Checked names: {{ checkedNames }}</div>
 
-  <input type="checkbox" id="demo-jack" value="Jack" v-model="checkedNames">
+  <input type="checkbox" id="demo-jack" value="Jack" v-model="checkedNames" />
```
Comment (yyx990803) [inline src/guide/essentials/forms.md:171]: Why are you adding the `/` in some cases but removing them in other places?

**Comment by dammy001** [comment]
@yyx990803 all updated.

File: src/guide/essentials/template-refs.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
 While Vue's declarative rendering model abstracts away most of the direct DOM operations for you, there may still be cases where we need direct access to the underlying DOM elements. To achieve this, we can use the special `ref` attribute:
 
 ```vue-html
-<input ref="input">
+<input ref="input" />
```
Comment (skirtles-code) [inline src/guide/essentials/template-refs.md:6]: Shouldn't this slash be removed?

File: src/api/ApiIndex.vue (hunk 1 above)
Comment (skirtles-code) [reply src/api/ApiIndex.vue:8]: I think I preferred the code for the template of `ApiIndex.vue` the way it was originally. There might be a small amount of duplication, but given the nested loops I think it's clearer to use, e.g. `section.text` rather than just `text`. It helps to differentiate `section.text` from `item.text`.

What is the motivation behind explicitly adding the type information?

**Comment by skirtles-code** [comment]
I'm going to close this.

The most important typos have been fixed in other PRs.

This PR currently has conflicts and can't currently be merged. There is also feedback that hasn't been addressed that would prevent it being merged.
